src/collection/backend/lmdb.cc

```diff
@@ -503,6 +506,61 @@ void LMDB::resolveRegularExpression(const std::string& var,
     return;
 }
 
+
+MDBEnvProvider* MDBEnvProvider::provider_ = nullptr;;
+
+MDBEnvProvider* MDBEnvProvider::GetInstance() {
+    if (provider_==nullptr) {
+        provider_ = new MDBEnvProvider();
+    }
+    return provider_;
+}
+
+void MDBEnvProvider::Finalize() {
+    if (provider_!=nullptr) {
+        provider_->close();
+        provider_ = nullptr;
+    }
+}
+
+MDBEnvProvider::MDBEnvProvider() :
+    m_env(NULL), initialized(false) {
+    pthread_mutex_init(&m_lock, NULL);
+}
+
+MDB_env* MDBEnvProvider::GetEnv() {
+    init();
+    return m_env;
+}
+
+MDB_dbi* MDBEnvProvider::GetDBI() {
+    init();
+    return &m_dbi;
+}
+
+void MDBEnvProvider::init() {
+    pthread_mutex_lock(&m_lock);
+    if (!initialized) {
```

Comment:
You forgot to change value of this `initialized` variable to `true` :)

Reply:
good point, thanks for catching that